src/Workspaces/Remote/Core/SolutionAssetStorage.cs

```diff
@@ -148,9 +135,11 @@ private void DisposeScope(Scope scope)
         }
 
         /// <summary>
-        /// Retrieve assets of specified <paramref name="checksums"/> available within <paramref name="scopeId"/> scope from the storage.
+        /// Retrieve assets of specified <paramref name="checksums"/> available within <paramref name="scope"/> from the
+        /// storage.
         /// </summary>
-        public async ValueTask<IReadOnlyDictionary<Checksum, SolutionAsset>> GetAssetsAsync(int scopeId, IEnumerable<Checksum> checksums, CancellationToken cancellationToken)
+        public static async ValueTask<IReadOnlyDictionary<Checksum, SolutionAsset>> GetAssetsAsync(
+            Scope scope, IEnumerable<Checksum> checksums, CancellationToken cancellationToken)
```

Comment:
note: tehse methods becoming static is very suspect in terms of where this code should live.  I could consider making it instance methods on Scope itself.